acquia/cli
# CLI-703: Use defined output for processes

**Motivation**

Our tests always include bogus output:
```
> phpunit tests/phpunit -vvv
PHPUnit 9.5.19 #StandWithUkraine

Runtime:       PHP 8.0.16 with PCOV 1.0.11
Configuration: /Users/dane.powell/src/acquia/cli/phpunit.xml.dist

    ✔ Testing!
...  63 / 198 ( 31%)
............................................................... 126 / 198 ( 63%)
............................................................... 189 / 198 ( 95%)
..cat: https://google.com: No such file or directory
.......                                                       198 / 198 (100%)
```

The presence of `cat: https://google.com: No such file or directory` and the checkmark indicates that some of our helpers aren't using the BufferedOutput provided by tests. They also point to a larger issue, which is that these helpers directly invoke a TTY, ignoring any provided output object, which could cause all kinds of unexpected behavior (such as hanging waiting for input during tests, *cough* #858).

We should always disable TTY during tests, but that's not possible today because it would also disable command output.

**Proposed changes**

Stop abusing Process::setTty(). Decouple TTY from command output by passing a callback to the Process to stream output regardless of TTY.

**Testing steps**


1. Follow the [contribution guide](https://github.com/acquia/cli/blob/master/CONTRIBUTING.md#building-and-testing) to set up your development environment.
2. Clear the kernel cache to pick up new and changed commands: `./bin/acli ckc`
3. Run `composer test`
4. Notice that `cat: https://google.com: No such file or directory` goes away
5. Run `wacli remote:drush foo.dev si` and verify you can still interact with the process (demonstrating TTY still works)

**Merge requirements**
